### DISCUSSION TITLE
[Docs] `watch` task without completion signal

### DISCUSSION THREAD

**Discussion by valtlai** [gulpjs/gulp#2476] [Help] [answered]
The [sample code](https://github.com/gulpjs/gulp#sample-gulpfilejs) in README contain the following Gulp task, which doesn’t signal async completion.

```js
function watch() {
  gulp.watch(paths.scripts.src, scripts);
  gulp.watch(paths.styles.src, styles);
}
```

When I run a very similar task, the console says:

```
[23:01:10] The following tasks did not complete: watch
[23:01:10] Did you forget to signal async completion?
```

Should the example code be updated or did I miss something?

**Comment by phated** [comment, marked as answer]
Why is your watch task ending? Watch tasks are meant to run forever until ctrl+c is used to end it which exits the program outside of that logging.

**Reply by valtlai** [reply]
I don’t know. Most of the time the task working perfectly but sometimes it just ends. Very hard to reproduce.

But yeah, you’re right and the docs don’t need updating.

**Reply by valtlai** [reply]
Maybe it’s gulpjs/glob-watcher#55 causing this issue. I’m using Node 14.

**Comment by dimaslanjaka** [comment]
my gulp watch did not completing the task too

```js
gulp.task('x', function (done) {
  console.log('x');
  done();
});
gulp.task('watch', function () {
  gulp.watch('./posts/**/*.md', { delay: 1000 }, gulp.series(['x']));
  setTimeout(() => {
    fs.appendFileSync(__dirname + '/posts\\hexo-theme-unit-test\\post-watcher.md', EOL);
  }, 1000);
});
```

![image](https://user-images.githubusercontent.com/12471057/210444259-f508d1c1-619f-4228-86b6-680625e08260.png)

i've been finding from google, but the answer is same. and i try to test, but not working. iam using `"gulp": "^4.0.2"`
